applications/PfemFluidDynamicsApplication/custom_strategies/strategies/two_step_v_p_strategy.h

```diff
@@ -828,14 +826,18 @@ class TwoStepVPStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinea
 	    }else{
 	      ElementalVolume = 0;
 	    }
-	    double CriticalVolume=-10;
-	    if(ElementalVolume<CriticalVolume){
-	      (itElem)->Reset(ACTIVE);
+	    double ModelPartVolume=ModelerUtils.ComputeModelPartVolume(rModelPart);
+	    double CriticalVolume=0.01*ModelPartVolume/double(rModelPart.Elements().size());
```

Comment:
I think we should find another definition of critical volume. One based on h maybe. I am thinking about the situation when we have adaptive or zone refinement. This is one of the strong points of PFEM vs SPH, for instance. For SPH it is very difficult to change the density of points across zones...